src/ILCompiler.ReadyToRun/src/Compiler/DependencyAnalysis/ReadyToRun/ExceptionInfoLookupTableNode.cs

```diff
@@ -0,0 +1,188 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using ILCompiler.DependencyAnalysisFramework;
+using Internal.Text;
+using Internal.TypeSystem;
+
+using Debug = System.Diagnostics.Debug;
+
+namespace ILCompiler.DependencyAnalysis.ReadyToRun
+{
+    class SyntheticSymbolNode : ISymbolDefinitionNode
+    {
+        MethodWithGCInfo _method;
+
+        ObjectNode.ObjectData _ehInfo;
+
+        public SyntheticSymbolNode(MethodWithGCInfo method, ObjectNode.ObjectData ehInfo)
+        {
+            _method = method;
+            _ehInfo = ehInfo;
+        }
+
+        public int Offset => 0;
+
+        public bool RepresentsIndirectionCell => false;
+
+        public bool InterestingForDynamicDependencyAnalysis => false;
+
+        public bool HasDynamicDependencies => false;
+
+        public bool HasConditionalStaticDependencies => false;
+
+        public bool StaticDependenciesAreComputed => true;
+
+        public bool Marked => true;
+
+        public void AppendMangledName(NameMangler nameMangler, Utf8StringBuilder sb)
+        {
+            sb.Append("SyntheticSymbolNode->");
+            _method.AppendMangledName(nameMangler, sb);
+        }
+
+        public IEnumerable<DependencyNodeCore<NodeFactory>.CombinedDependencyListEntry> GetConditionalStaticDependencies(NodeFactory context)
+        {
+            return Array.Empty<DependencyNodeCore<NodeFactory>.CombinedDependencyListEntry>();
+        }
+
+        public IEnumerable<DependencyNodeCore<NodeFactory>.DependencyListEntry> GetStaticDependencies(NodeFactory context)
```

Comment:
Nit: we tend to just return null from these and group them near the end of the file. The perf improvement won't be measurable, but it's less code.

https://github.com/dotnet/corert/blob/635cf21aca11265ded9d78d216424bd609c052f5/src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/EmbeddedObjectNode.cs#L61-L66

Reply:
Or wait... these are not overrides because this class is not a dependency node. Why do we need these methods?

Reply:
Hmm, I believe that SyntheticSymbolNode is an ISymbolDefinitionNode which is an IDependencyNode<NodeFactory> - at least the compiler claims so when I try to delete one of the methods. I have put them to the end of the class and reformatted them based on your suggestion though, that is a good organizational suggestion.